src/hooks/permissionedPools/PermissionedPositionManager.sol

```diff
@@ -70,10 +71,33 @@ contract PermissionedPositionManager is PositionManager {
         bytes calldata hookData
     ) internal override {
         // allowlist is verified in the hook call
-        if (poolKey.hooks != PERMISSIONED_HOOKS) revert InvalidHook();
+        if (!_checkAllowedHooks(poolKey)) revert InvalidHook();
         super._mint(poolKey, tickLower, tickUpper, liquidity, amount0Max, amount1Max, owner, hookData);
     }
 
+    function _checkAllowedHooks(PoolKey calldata poolKey) internal view returns (bool) {
+        if (
+            !_checkAllowedHook(poolKey.currency0, poolKey.hooks) || !_checkAllowedHook(poolKey.currency1, poolKey.hooks)
+        ) {
+            return false;
+        }
+        return true;
+    }
+
+    function _checkAllowedHook(Currency currency, IHooks hooks) internal view returns (bool) {
+        address permissionedToken = _verifiedPermissionedTokenOf(currency);
+        if (permissionedToken == address(0)) return true;
+        return isAllowedHooks[currency][hooks];
+    }
+
+    /// @dev Sets which hooks are allowed to be used with a wrapped permissioned token. Only callable by the owner of the wrapped permissioned token
+    function setAllowedHook(Currency currency, IHooks hooks, bool allowed) public {
```

Comment:
function order should follow the solidity style guide:
- external
- public
- internal
- private

also the function should be external